### PR TITLE
Documentation update in respect to code fragments

### PR DESCRIPTION
**Pull request by albert-github**
Update of documentation in respect of code fragments
- making code fragments work in footnotes
- placing some words in in code tags
- completing some words to code tags
- removing some `#` as they are not necessary
- don't show obfuscation code in email

### CODE REVIEW

File: Documentation/doc/Documentation/Developer_manual/Chapter_code_format.txt [hunk 1]
```diff
@@ -243,13 +243,13 @@ The first list of items are meant as rules, <I>i.e.</I>, you should follow them.
   modify the object to which it is applied,
   <I>e.g.</I>, `class A { int f( void) const; };`. This should also be
   done when it is only conceptually `const`.
-  This means that the member function `f()` is `c`onst as seen from
+  This means that the member function `f()` is `const` as seen from
   the outside, but internally it may modify some data members
-  that are declared `m`utable. An example
+  that are declared `mutable`. An example
   is the caching of results from expensive computations.  For more
   information about conceptually `c`onst functions and mutable data
```
Comment (GilesBathgate) [inline Documentation/doc/Documentation/Developer_manual/Chapter_code_format.txt:250]: was the fix to replace \`c\`onst with \`const\`? in which case there is another here.

Comment (albert-github) [reply Documentation/doc/Documentation/Developer_manual/Chapter_code_format.txt]: @GilesBathgate Thanks for your response.

Do you mean the line 250?

Comment (GilesBathgate) [reply Documentation/doc/Documentation/Developer_manual/Chapter_code_format.txt]: > Do you mean the line 250?

Yes.

**Comment by albert-github** [comment]
Any possibility to get this tested and integrated?

File: Documentation/doc/Documentation/Developer_manual/Chapter_portability.txt [hunk 2]
```diff
@@ -304,7 +304,7 @@ up-to-date version of this list.
 \subsection secworkaround_macros Macros connected to workarounds/compilers
 
 Some macros are defined according to certain workaround flags. This is
-done to avoid some \#`ifdef`s in our actual code.
+done to avoid some `#ifdef`s in our actual code.
```
Comment (sloriot) [inline Documentation/doc/Documentation/Developer_manual/Chapter_portability.txt:307]: This results in a warning:
`Documentation/Developer_manual/Chapter_portability.txt:307: warning: explicit link request to 'ifdef' could not be resolved`

Comment (albert-github) [reply Documentation/doc/Documentation/Developer_manual/Chapter_portability.txt:307]: Strange, with which version of doxygen?
When I run nmake Documentation_internal_doxygen_run or nmake doc with the current doxygen master (1.9.5 (86fe8ce908bfd6782633d2d9647f2159c2f08a44)) I don't get a warning.
With the overnight build [CGAL-5.5-Ic-80](https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.5-Ic-80/index.html) I also see no indication of it.

Comment (albert-github) [reply Documentation/doc/Documentation/Developer_manual/Chapter_portability.txt:307]: I did a small text with just a small text and used
```
some `#``ifdef`s in
```
and with the small test the 1.8.4, 1.8.13 and the current master give no warning and the output looks OK
(the internal code will look like  `some <code>#</code><code>ifdef</code>s in`, which is not nice but is OK as well).

When you see the problem with the 1.8.4 or the 1.8.13 version, we might use this version of the code.
(Though this is also a nice reminder of going to a newer version of doxygen  #5840)

Comment (sloriot) [inline Documentation/doc/Documentation/Developer_manual/Chapter_portability.txt:307]:
```suggestion
done to avoid some `ifdef`s in our actual code.
```

Comment (albert-github) [reply Documentation/doc/Documentation/Developer_manual/Chapter_portability.txt:307]: This is also good suggestion and will probably not really disturb the readability.

**Pull request merged**